core/include/kernel/tee_misc.h

```diff
@@ -83,6 +83,15 @@ bool _core_is_buffer_intersect(vaddr_t b, size_t bl, vaddr_t a, size_t al);
 
 /*  UUID string: "XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXXXXXX". Size includes '\0' */
 #define TEE_UUID_STRING_LEN		36
-int uuid2str(char *dst, TEE_UUID *uuid);
+int uuid2str(char *dst, const TEE_UUID *uuid);
+
+static inline char *_uuid2str(const TEE_UUID *uuid)
+{
+	static char str[TEE_UUID_STRING_LEN];
```

Comment:
Option 3 would be nice, as long as we don't loose the option to use
`__attribute__ ((__format__ (__printf__, 3, 4)))`
for `snprintf()` and friends that's the best choice. Second best is option 2.

With option 3 we would probably need to cast a `TEE_UUID *` to a `void *` when using `%pU`, but that's not that bad.

I don't think it's that difficult to hack that into kprintf, looks like an interesting challenge. :-)